test/java/com/arpnetworking/metrics/portal/integration/controllers/ReportControllerIT.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2019 Dropbox Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.integration.controllers;
+
+import com.arpnetworking.commons.jackson.databind.ObjectMapperFactory;
+import com.arpnetworking.metrics.portal.integration.test.WebServerHelper;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Charsets;
+import com.google.common.io.Resources;
+import models.view.reports.Report;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.message.BasicHeader;
+import org.junit.Test;
+import play.mvc.Http;
+
+import java.io.IOException;
+import java.net.URL;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Integration tests for {@code ReportController}.
+
+ * @author Spencer Pearson (spencerpearson at dropbox dot com)
+ */
+public final class ReportControllerIT {
+
+    @Test
+    public void testCreateValid() throws IOException {
+        HttpPut putRequest = new HttpPut(WebServerHelper.getUri("/v1/reports"));
+        putRequest.setHeader(new BasicHeader(HttpHeaders.CONTENT_TYPE, "application/json"));
+        putRequest.setEntity(new StringEntity(loadResource("testCreateValid")));
```

Comment:
The problem with putting a fixed resource is that the test cannot be run again on the same instance because the data will collide. As per comments in AlertControllerIT we should move to using randomly generated data instead of JSON fixtures. There is already support for many Ebean and internal models in TestBeanFactory and I see no reason not to extend support for view model creation.

Reply:
^ Any specific fields that the test needs to set for its condition should be explicitly set in the test. This way the data that relates to the test is clearly spelled out in the test (it's very non-obvious reading JSON what the relevant part of the JSON blob is to a particular test).

Reply:
I opted to `models.view.impl.Report.fromInternal(TestBeanFactory.createReportBuilder())`. Good? Or is there a reason to have a separate view-model-creation method?